woocommerce/woocommerce-ios
# Jetpack CP: add `jetpackConnectionActivePlugins` attribute to `Site` entity for analytics

Part of #5402 


Notes: 742 diffs are on the new model version!

### Description


For JCP analytics p91TBi-6r6-p2, we want to know which active plugins are on JCP sites and we can fetch this property in `me/sites`' option `jetpack_cp_active_plugins`. The use case is `site_picker_continue_tapped`, when the user selects a site in the site picker and continues with the site. Even though this property is only for analytics, it is much easier to persist this in `Storage.Site` since we use fetched results controller in the site picker. Persisting this property also makes its value more up-to-date, in case the user selects the JCP site while `me/sites` request has not returned. WCAndroid also persists this attribute in https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2180.

### Changes

This PR only contains the storage layer changes for easier review:

- Added Core Data model version 59 which adds `jetpackConnectionActivePlugins` attribute to `Site` entity with a test case on the default value (`nil`) and setting the attribute

### Testing instructions


Feel free to launch the app to make sure everything runs as before after model migration!

---
